pilot/cmd/pilot-agent/main.go

```diff
@@ -346,25 +346,25 @@ var (
 			// Legacy - so pilot-agent can be used with citadel node agent.
 			// Main will be replaced by istio-agent when we clean up - this code can stay here and be removed with the rest.
 			sdsUDSPath := sdsUdsPathVar.Get()
-			sdsEnabled, sdsTokenPath := detectSds(controlPlaneBootstrap, sdsUDSPath, trustworthyJWTPath)
+			nodeAgentSDSEnabled, sdsTokenPath := detectSds(controlPlaneBootstrap, sdsUDSPath, trustworthyJWTPath)
 
-			if !sdsEnabled && role.Type == model.SidecarProxy { // Not using citadel agent - this is either Pilot or Istiod.
+			if !nodeAgentSDSEnabled && role.Type == model.SidecarProxy { // Not using citadel agent - this is either Pilot or Istiod.
 
 				// Istiod and new SDS-only mode doesn't use sdsUdsPathVar - sdsEnabled will be false.
 				sa := istio_agent.NewSDSAgent(discoveryAddress, controlPlaneAuthEnabled)
 
-				if sa.JWTPath != "" && role.Type == model.SidecarProxy {
```

Comment:
applies to gateways now?

Reply:
this was always intended to be used with gateways, they check was added due to some test failures that were occuring and have since been fixed

Reply:
Someone will need to switch the gateway too..